### PR TITLE
Fix missing free() in $INCLUDE

### PR DESCRIPTION
**Pull request by jpereira**

### CODE REVIEW

File: src/lib/server/cf_file.c [hunk 1]
```diff
@@ -800,22 +800,21 @@ static int process_include(CONF_SECTION *this, char const *ptr, char *buff[stati
 		if (stat(my_directory, &stat_buf) < 0) {
 			ERROR("%s[%d]: Failed reading directory %s: %s", filename, *lineno,
 			      my_directory, fr_syserror(errno));
-			talloc_free(my_directory);
-			return -1;
+			goto error;
```
Comment (arr2036) [inline src/lib/server/cf_file.c]: Goto label should be here

File: src/lib/server/cf_file.c [hunk 2]
```diff
@@ -800,22 +800,21 @@ static int process_include(CONF_SECTION *this, char const *ptr, char *buff[stati
...
 		}
 #endif
 		dir = opendir(my_directory);
 		if (!dir) {
 			ERROR("%s[%d]: Error reading directory %s: %s",
 			      filename, *lineno, value,
 			      fr_syserror(errno));
+		error:
```
Comment (arr2036) [inline src/lib/server/cf_file.c]: not in the middle of the cases where the error could occur

File: src/lib/server/cf_file.c [hunk 3]
```diff
@@ -853,8 +852,7 @@ static int process_include(CONF_SECTION *this, char const *ptr, char *buff[stati
 			 */
 			if (cf_file_include(this, buff[2], CONF_INCLUDE_FROMDIR, buff, true) < 0) {
 				closedir(dir);
-				talloc_free(my_directory);
-				return -1;
+				goto error;
```
Comment (arr2036) [inline src/lib/server/cf_file.c]: just squash the commits...

**Comment by jpereira** [comment]
done @arr2036

**Comment by arr2036** [comment]
Goto label still not at the first incidence of the error.

-Arran

**Comment by jpereira** [comment]
As I said, take a look at the code and you will see that the first incidence is inside a `#ifdef ... #endif` condition. either way, I just fixed the missing `free()` and kept as the old approach.

**Comment by arr2036** [comment]
@jpereira you didn't say that.  Or at least I didn't get a notification of you saying that...

**Pull request merged**